source/module_base/test/complexarray_test.cpp

```diff
@@ -45,125 +50,419 @@
 *   - operator "*=":
 *       similar as "*"
 *
+*   - operator "==":
+*       judge if two ComplexArray is equal, all element is equal.
+*
+*   - operator "!=":
+*       judge if two ComplexArray is not equal
+*
 *   - oprator "()":
 *       - access the element
+*
+*   - function zero_out, negate, randomize, getBound/Size, create.
+*
+*   Test relative functions:
+*   - overloading of operator "*". a double/complex multiply a Class ComplexArray.
+*   - functon abs2(), return the sum of squares of all elements 
+*   - function dot()
+*   - function scale_accumulate()
+*   - function scaled_sum()
+*   - function point_mult()
 */
 
 
+double prec_test = 1e-14;
```

Comment:
You may remove redundant codes and variables: line 74, 78-79.

Reply:
ok